test/functional/lua/treesitter_spec.lua

```diff
@@ -404,4 +404,57 @@ static int nlua_schedule(lua_State *const lstate)
     end
     eq({true,true}, {has_named,has_anonymous})
   end)
+  it('allows to set ranges', function()
+    if not check_parser() then return end
+
+    insert(test_text)
+
+    local res = exec_lua([[
+    parser = vim.treesitter.get_parser(0, "c")
+    return { parser:parse():root():range() }
+    ]])
+
+    eq({0, 0, 19, 0}, res)
+
+    local res = exec_lua([[
+    local root = parser:parse():root()
+    parser:set_included_ranges({{root:child(0), root:child(0)}})
```

Comment:
Can you give just a little info about this?

Also, can you split them into individual `it` tests, so you can say clearly what each is doing?

Reply:
Yes, it should be done now !

Reply:
@vigoux and I were discussing the API of `set_included_ranges` which is now called as an array of tuple `{ {node_start, node_end},  .., {node_start, node_end}` . I was triggered by the use of the same node here `root:child(0), root:child(0)}` and proposed to either pass a list of ranges like `set_included_ranges( {create_range(node_start,node_end),  create_range(node_start2,node_end2)} )` or just pass a list of node and use their ranges as done in https://tree-sitter.github.io/tree-sitter/using-parsers#multi-language-documents. @tjdevries what do you think ?

Reply:
As discussed, we are going for a `table of nodes` solution, as it seems to be simpler on both implementation and usage. More complex ranges will in any case be easy to generate by iterating over each intermediate node between two nodes.
Something else to note is that this is used mainly for language injection, and most of the parsers, have clearly identified nodes that needs language injection on them.